client/src/model/component/CircuitDiagram.hx

```diff
@@ -215,4 +215,124 @@ class CircuitDiagram extends Observable implements CircuitDiagramI implements Ob
         }
         return worldPointArray;
     }
+	
+	public function normalise() : Void {
+		// For all pairs of connections {c0, c1} such that c0 != c1 and c0 is close to c1
+		// if and both contain a port,
+			// create a link to connect them
+		// else one or both don't contain a port
+			// if c0 contains a port, swap c0 and c1
+			// move everything in c0 to c1
+		var connectionArray = new Array<Connection>();
+		for (link in this.linkArray) {
+			connectionArray.push(link.get_endpoint(0).getConnection());
+			connectionArray.push(link.get_endpoint(1).getConnection());
+		}
+		
+		for (component in this.componentArray) {
+			var ports = component.get_ports();
+			for (port in ports) {
+				connectionArray.push(port.getConnection());
+			}
+		}
+		
+		var markedConnections = new Array<Connection>();
+		for (connection in connectionArray) {
+			var otherConnections = connectionArray;
+			otherConnections.remove(connection);
+			for (otherConnection in otherConnections) {
+				var connectionDistance = Link.pointsDistance(connection.get_xPosition(), connection.get_yPosition(), otherConnection.get_xPosition(), otherConnection.get_yPosition());
+				if (connectionDistance <= 15) {
+					// here 15 is what is being considered as "close" to each other
+					if (connection.aPortIsConnecte() && otherConnection.aPortIsConnecte()) {
+						this.addLink(new Link(this, connection.get_xPosition(), connection.get_yPosition(), otherConnection.get_xPosition(), otherConnection.get_yPosition()));
+					} else {
+						if (connection.aPortIsConnecte()) {
+							for (element in connection.get_connectedElements()) {
+								otherConnection.connect(Std.downcast(element, Connectable));
+							}
+							markedConnections.push(connection);
+						}
+					}
+					
+					
+				} 
+			}
+		}
+		
+		// removing marked connections. Unclear is swapping connections in use cases is actually what it is supposed to be.
+		// removing such unused connections makes more sense. CHECK ?
+		for (connection in markedConnections) {
+			connectionArray.remove(connection);
+		}
```

Comment:
This loop is looks fine to me.  The marked connections should no longer be a part of the circuit diagram.

Another way to achieve the same thing is to go back to all the links and ports and gather all the connections from them.  In either case you should arrive at the same set of connection.  One caveat, though, if you are using arrays to represent sets and there are duplicates, then remove will only remove the first occurrence.  Better to make a proper Set type.